modules/components/src/AdvancedSqonBuilder/filterComponents/BooleanFilter.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Component from 'react-component-component';
+
+import {
+  getOperationAtPath,
+} from '../utils';
+import BooleanAgg from '../../Aggs/BooleanAgg';
+import { FilterContainer } from './common';
+import './FilterContainerStyle.css';
+
+const getFieldDisplayName = (fieldDisplayNameMap, initialFieldSqon) => {
+  return fieldDisplayNameMap[initialFieldSqon.content.field] || initialFieldSqon.content.field;
+};
+
+const AggsWrapper = ({ children }) => (
+  <div className="aggregation-card">{children}</div>
+);
+
+export const BooleanFilterUI = props => {
+  const {
+    onSubmit = sqon => {},
+    onCancel = () => {},
+    ContainerComponent = FilterContainer,
+    sqonPath = [],
+    initialSqon = null,
+    field = '',
+    fieldDisplayNameMap = {},
+  } = props;
+
+  const initialState = {
+    selectedValue: undefined,
+    localSqon: initialSqon
+  };
+
+  const initialFieldSqon = getOperationAtPath(sqonPath)(initialSqon) || {
+    op: 'and',
+    content: { value: initialState.selectedValue, field },
+  };
+
+  const onSqonSubmit = s => () => onSubmit(s.state.localSqon);
+
+  const onSelectionChange = s => ({ value, generateNextSQON }) => {
+    s.setState({
+      selectedValue: value.key_as_string,
+      localSqon: generateNextSQON(s.state.localSqon),
+    });
+  };
+
+  const isActive = s => ({ value }) => value === s.state.selectedValue;
+
+  return (
+    <Component initialState={initialState}>
+      {s => (
+        <ContainerComponent onSubmit={onSqonSubmit(s)} onCancel={onCancel}>
+          <React.Fragment>
+            <div key="header" className="contentSection">
+              <span>
+                { `Participant is a ${getFieldDisplayName(fieldDisplayNameMap, initialFieldSqon)}`}
```

Comment:
Interesting problem we have here. We don't want to have domain specific concepts embedded here, so this string will not work, but I will speak with Kim to see if we can come up with a more suitable language given the constraint :) 

Just FYI, `fieldDisplayNameMap` is meant to be something like: `{["participant.proband"] : "Proband"}`, so it represents a display name for the field, not the value of the field ;)

Will keep you in the loop :)

Reply:
That's an issue I've seen elsewhere too : we inject the human readable name of a field in a component, only to inject it in a sentence that contains specific language.
I think we need a broad solution, not one specific to this case.